test/functional/p2p_initial_headers_sync.py

```diff
@@ -80,7 +76,6 @@ def run_test(self):
                 if "getheaders" in p.last_message:
                     count += 1
                     peer_receiving_getheaders = p
-                    p.last_message.pop("getheaders", None)
```

Comment:
don't think we can remove this since we need to clear initial getheaders in peer2, peer3 before using `wait_for_getheaders` the first time so that it detects new getheaders.

Reply:
We are asserting this is clean between connecting and announcing the block, so I think this should be safe:
https://github.com/bitcoin/bitcoin/blob/80ec483a3b0dfb9e43c06b099eb272813739bd54/test/functional/p2p_initial_headers_sync.py#L60-L62

Reply:
inserting an `assert "getheaders" not in p.last_message` fails here because it received a new header after those checks. and the block hash doesn't change here.

Reply:
Sure, because "getheaders" is in `last_message` for that peer, but only for that one as the assert after the loop is checking. 

The later `wait_for_getheaders` (the one on `expected_peer`) is checked on the other node only. I guess that is not too clear based on how `expected_peer` is initialized. We could just set `expected_peer` as follows to make it clearer:

```python
for p in [peer2, peer3]:
    with p2p_lock:
        if "getheaders" in p.last_message:
            count += 1
            p.send_message(msg_headers()) # Send empty response, see above
        else:
            expected_peer = p
```

Reply:
oh makes sense! `wait_for_getheaders` is done on the peer which we made sure didn’t receive getheaders before. i'm fine with leaving it as it is.